### PR TITLE
nc news6

### PR DESCRIPTION
**Pull request by YoriB**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
+    const article_id = 3;
     return request(app)
-    .get('/api/comments/2/count')
+    .get(`/api/comments/${article_id}`)
     .expect(200)
-    .then(({body}) => {
-      console.log(body)
-    })
+    .then(({body}) => { 
+     for (let i = 0; i < body.length; i++) {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:121]: this for loop is completely unnecessary to use with a length test inside, but you might test the length first and then forEach through each comment that is contains the expected properties

File: __tests__/app.test.js [hunk 2]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
     return request(app)
-    .get('/api/comments/2/count')
+    .get(`/api/comments/${article_id}`)
     .expect(200)
-    .then(({body}) => {
-      console.log(body)
-    })
+    .then(({body}) => { 
+     for (let i = 0; i < body.length; i++) {     
+     expect(body.length).toEqual(2);
```
Comment (alexdevdavis) [inline __tests__/app.test.js:122]: this is a good test to use before a forEach

File: __tests__/app.test.js [hunk 3]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
+  
+  test('400 status response with an  invalid article ID', () => {
+    return request(app)
+    .get('/api/comments/banaanaz')
+    .expect(400)
+    .then(({body})=> {     
+      expect(body.msg).toEqual('Bad request');
+    })   
+  })
+
```
Comment (alexdevdavis) [inline __tests__/app.test.js:173]: nice error handling tests

File: models/model.js [hunk 4]
```diff
@@ -37,22 +37,48 @@ const fetchArticlesById = (article_id) => {
...
+    FROM articles 
+    WHERE article_id = $1`, [article_id])
 
     .then((result) => {
       return result.rows[0];
     });
 };
 
-const fetchCommentsByArticleId = (article_id) => {
+// const fetchCommentsByArticleId = (article_id) => {
```
Comment (alexdevdavis) [inline models/model.js:49]: avoid committing and pushing commented out code

File: test.sql [hunk 5]
```diff
@@ -1,27 +1,5 @@
 \c nc_news_test
 
+SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;
```
Comment (alexdevdavis) [inline test.sql:4]: I suggest you add this file to your .gitignore

File: test.txt [hunk 6]
```diff
@@ -1,17 +1 @@
 You are now connected to database "nc_news_test" as user "northcoders".
```
Comment (alexdevdavis) [inline test.txt:1]: and gitignore this one too

**Review by alexdevdavis — APPROVED** [review]
Nice one, Yori. Happy to approve this. 
- refactor suggestions in comments

File: __tests__/app.test.js (hunk 1 above)
Comment (YoriB) [reply __tests__/app.test.js:121]: OK will remove the loop.

**Pull request merged**